packages/frontend-shared/src/composables/useAddProject.ts

```diff
@@ -0,0 +1,37 @@
+import { gql } from '@urql/core'
+import { useMutation } from '@urql/vue'
+
+function makeAddProjectQuery (query?: string) {
+  const projectsQuery = `
+    projects {
+      id
+      title
+      projectId
+      projectRoot
+      isFirstTimeCT
+      isFirstTimeE2E
+    }
+  `
+
+  return gql`
+    mutation addProject($path: String!) {
+      addProject(path: $path) {
+        ${query ? query : projectsQuery}
```

Comment:
If we do go with composables, this is an anti-pattern (field interpolation in the `gql` template literal) - it breaks codegen. If we needed to do things conditionally we could consider `@include` / `@skip` directives, but I think it's probably best to keep things simple for now